### DISCUSSION TITLE
`next build` should respect `rootDir` in tsconfig to prevent redundant type-checking

### DISCUSSION THREAD

**Discussion by ctjlewis** [vercel/next.js#27520] [Ideas]
### Describe the feature you'd like to request

`next build` already respects tsconfig's `exlcude` field, but it does not do so for the `rootDir` field, resulting in users having to manually exclude any folders in their project directory that they do not want the build process to type-check.

Adding support for `rootDir` would eliminate the need for this workaround and prevent redundant type-checking in general.

See:
- https://github.com/vercel/next.js/issues/11144
- https://github.com/vercel/next.js/discussions/11113

### Describe the solution you'd like

I believe the best approach would to enable Next to respect `rootDir` like it respects `exclude`:

> My workaround for this was adding the redundant folder to `exclude` in tsconfig.json, though if Next respected the `rootDir` field, the workaround wouldn't be necessary.
> 
> If Next respects `"rootDir": "src"`, then it will know to ignore everything outside of `src/` in this example, and `nextDontCompile/` could be compiled by a separate process with its own local tsconfig (even using `"extends": "../tsconfig.json"`):
> 
> ```
> .
> ├── nextDontCompile
> │   ├── index.tsx
> │   └── tsconfig.json
> ├── src
> │   ├── index.tsx
> │   └── utils.ts
> └── tsconfig.json
> ```

See: https://github.com/vercel/next.js/discussions/11113#discussioncomment-1052618_

### Describe alternatives you've considered

You can manually ignore the folders you don't want `next build` to redundantly type-check by adding the path to `exclude` in tsconfig, but this is not scalable for a large number of folders.

**Comment by timneutkens** [comment]
Didn't take a deeper look into this yet but we use the TypeScript programmatic API to read the configuration file in your project so it should pass along the options to the checker. Relevant code is here if you want to have a look: https://github.com/vercel/next.js/blob/canary/packages/next/lib/typescript/runTypeCheck.ts#L25

**Reply by ctjlewis** [reply]
Thank you Tim. I can commit to having this PR in by EOW if someone else is willing to write the tests.

**Reply by ctjlewis** [reply]
Also I did not mean to open this as an issue instead of a discussion - there is still a template up for feature requests under issues, so I mistakenly went for that.

**Reply by timneutkens** [reply]
> Also I did not mean to open this as an issue instead of a discussion - there is still a template up for feature requests under issues, so I mistakenly went for that.

That's the right flow currently, the problem we have is that discussions do not support templates so we have people create a feature request using the issue template form and then convert it to discussions.
